src/feature/dircache/dircache.c

```diff
@@ -1456,13 +1456,19 @@ handle_get_next_bandwidth(dir_connection_t *conn,
     char *bandwidth = read_file_to_str(options->V3BandwidthsFile,
                                        RFTS_IGNORE_MISSING, NULL);
     if (bandwidth != NULL) {
-      size_t len = strlen(bandwidth);
-      write_http_response_header(conn, len, compress_method,
-                                 BANDWIDTH_CACHE_LIFETIME);
-      if (compress_method != NO_METHOD)
+      ssize_t len = strlen(bandwidth);
+      write_http_response_header(conn, compress_method != NO_METHOD ? -1 : len,
+                                 compress_method, BANDWIDTH_CACHE_LIFETIME);
+      if (compress_method != NO_METHOD) {
         conn->compress_state = tor_compress_new(1, compress_method,
-                                        choose_compression_level(len));
-      connection_buf_add(bandwidth, len, TO_CONN(conn));
+                                        choose_compression_level(len/2));
+        log_debug(LD_DIR, "Calling connection_buf_add_compress.");
```

Comment:
This log message should probably have a slightly different string to it. Something like "Compressing bandwidth cache file"